org.eclipse.search/search/org/eclipse/search/internal/ui/SearchPreferencePage.java

```diff
@@ -91,6 +92,9 @@ public static void initDefaults(IPreferenceStore store) {
 		store.setDefault(TEXT_SEARCH_ENGINE, ""); //default search engine is empty string //$NON-NLS-1$
 		store.setDefault(TEXT_SEARCH_QUERY_PROVIDER, ""); // default query provider is empty string  //$NON-NLS-1$
 		store.setDefault(LIMIT_HISTORY, 10);
+
+		// we will use 0 to mean disable maxResults limiting functionality
+		store.setDefault(MAX_RESULTS, 0);
```

Comment:
I would prefer a good default. 100_000 seems to be a good compromise when problems start at 200_000

Reply:
It seems like some people have 1,000,000 starting point desires and given that's not my usage scenario... I can't really contribute to that argument discussion, so I think I need to let those people decide between themselves and this can be changed here when the decision is made by everyone.

There's also the performance regression problem some one pointed out. By disabling it by default it would at least make sure no one gets a regression.

It's not my preferred option as I think a small default will provide the best out-of-the-box user experience for other people; but given the disagreement and regression, I think this at least helps this move forward for now.

Reply:
Note, the performance problem was fixed yesterday via #189.